### DISCUSSION TITLE
RFE: UI customization

### DISCUSSION THREAD

**Discussion by BrianHanke** [AcademySoftwareFoundation/OpenRV#181] [Ideas]
Hi all, just a quick follow-up to the meeting this morning. I think these are some of the things that would be nice to see in Open RV, UI-wise:

1. Skin/theme chooser in-app, with user-submitted themes available for download.
2. Theme designer GUI for choosing colors, fonts.
3. Sample stylesheet with annotations explaining what everything does.
4. New icons from flaticon.com or elsewhere.
5. The Annotation icons are not hardcoded, but the main window ones are. Change that so it's possible to swap icons without doing a full recompile?

Thanks!

**Comment by BrianHanke** [comment]
Here's the stylesheet I made. I tried to organize it clearly and I put notes about what various stuff does. Not sure it's 100% complete, but I tested quite a bit and it seems to cover everything. There are a couple links to icons that only work on my system.

[rv_stylesheet.zip](https://github.com/AcademySoftwareFoundation/OpenRV/files/11857526/rv_stylesheet.zip)

**Comment by bernie-laberge** [comment]
Hello @BrianHanke,
I apologize it took so long to follow this up with you ! I know it's been two years but we would be really interested in assisting you in developing your UI Customization/Improvements ideas further. Salomon (@salolavyadsk) is our RV summer intern at Autodesk and we were wondering if he could assist you in addressing some of the excellent ideas you had. Perhaps we could start by updaing the default stylesheets and then look at the other ideas above.

**Comment by BrianHanke** [comment]
Thanks for the message, @bernie-laberge! I use my custom RV UI every day. It's pretty much "done" in that I'm happy with it and haven't changed anything in a long time. The mods can be grouped into the following sections:

1. Stylesheet.
2. Icons from some free icon site that I can't remember right now (can find it later.)
3. A change to qtcolortriangle.cpp to make it have a nicer background.
4. Updates to how the "cartouches" are drawn.
5. Menu reorganization.

Feel free to keep in touch and let me know how you'd like to proceed. I'm available and happy to help. 

Best,

Brian

**Comment by salolavyadsk** [comment]
Hi @BrianHanke,

It’s great to meet you! I’m Salomon, the new RV intern. Yesterday, I began reviewing the current UI and stylesheet, and I really appreciate your vision for modernizing the interface — it’s exciting to see the direction you’ve laid out.
I saw your previous suggestions for improvement, including:

1. Updating the stylesheet
2. Introducing new icons
3. Adding an in-app skin/theme chooser
4. Creating a theme designer GUI for customizing colors and fonts

As of now, we’re starting with the stylesheet and icon updates, and then plan to move on to implementing the skinning system and theme designer.
If you have some time, I’d love to schedule a call to hear your insights and tips on how to make this a reality. Your input would be 
incredibly helpful!

Best regards,
Salomon

**Comment by BrianHanke** [comment]
Hi @salolavyadsk, sure, drop me a line any time to set something up. My email address is brianrhanke at outlook.
